lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -485,17 +513,20 @@ WasmBytecodeGenerator::EmitBlock()
     if (m_reader->IsBinaryReader())
     {
         UINT blockCount = m_reader->m_currentNode.block.count;
-        if (blockCount <= 0)
+        if (blockCount > 0)
         {
-            throw WasmCompilationException(_u("Invalid block node count"));
+            for (UINT i = 0; i < blockCount - 1; i++)
+            {
+                EmitInfo tmpInfo = EmitExpr(m_reader->ReadFromBlock());
+                ReleaseLocation(&tmpInfo);
+            }
+            // block yields last value
+            blockInfo = EmitExpr(m_reader->ReadFromBlock());
         }
-        for (UINT i = 0; i < blockCount - 1; i++)
+        else
         {
-            EmitInfo tmpInfo = EmitExpr(m_reader->ReadFromBlock());
-            ReleaseLocation(&tmpInfo);
+            blockInfo = EmitInfo(Wasm::WasmTypes::Void);
```

Comment:
Hmm, so an empty block with no count is a legal form or perhaps is blockCount being used for something else?